### ISSUE TITLE
Simplifications to how the grammar is presented?

### ISSUE THREAD

**Issue by mattfenwick** [H2CO3/Sparkling#16]
Looking at the [grammar](https://github.com/H2CO3/Sparkling/blob/master/doc/grammar.txt), I noticed a few things that I usually do differently -- which, for me, make the grammar shorter, clearer, easier to maintain and extend, and easier to use as a reference for implementing a parser (although that may depend on the choice of parsing technology).

I don't want to tell you how to do your business, but if you're interested, I would be happy to create a fork and whip up a full example.

Here are some of the issues I noticed (which are just my humble, often incorrect opinion):
- the regex quantifiers `rule(?)` and `rule(*)` are easier to read than `[ rule ]` and `{ rule }`
- these are hard for me to read:
  
  ```
  argument-declaration-list = IDENT { ',' IDENT }
  const-statement = 'const' IDENT '=' expression { ',' IDENT '=' expression } ';'
  ```
  
  these are much easier for me to read, since they get rid of the repetition, clearly show the separator and what's being separated (and how many):
  
  ```
  argument-declaration-list = sepBy1 ',' IDENT
  const-statement = 'const'  sepBy1( ','  init )  ';'
  init =  IDENT '=' expression
  ```
- specifying operator precedence is too complicated for me to understand; I find tables of precedence levels and associativities much easier to grasp, for example:
  
  ```
  {'left': {'+': 10, '-': 10, '*': 20, ...}, 
   'prefix': {'++': 100, ...}, ...}
  ```
- plus I noticed the tokens don't have grammar-ish definitions

So, in summary: none of these proposals change the language in any way, just how it's specified (with the goal of making the specification clearer).  

Feel free to close this issue and ignore me if you think it's a waste of time, or stupid, or whatever.
